src/game/common/skirmishbattlehonors.cpp

```diff
@@ -27,3 +39,617 @@ int SkirmishBattleHonors::Get_Endurance_Medal(Utf8String name, int level) const
     str.Format("%s_%d", name.Str(), level);
     return Get_Int(str, 0);
 }
+
+int SkirmishBattleHonors::Get_Wins() const
+{
+    Utf8String str = "Wins";
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_Win_Streak() const
+{
+    Utf8String str = "WinStreak";
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_Best_Win_Streak() const
+{
+    Utf8String str = "BestWinStreak";
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_Losses() const
+{
+    Utf8String str = "Losses";
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_China_Campaign_Complete(int level) const
+{
+    Utf8String str;
+    str.Format("CHINACampaign_%d", level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_USA_Campaign_Complete(int level) const
+{
+    Utf8String str;
+    str.Format("USACampaign_%d", level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_GLA_Campaign_Complete(int level) const
+{
+    Utf8String str;
+    str.Format("GLACampaign_%d", level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_Challenge_Campaign_Complete(int general, int level) const
+{
+    Utf8String str;
+    str.Format("ChallengeCampaign%d_%d", general, level);
+    return Get_Int(str, 0);
+}
+
+int SkirmishBattleHonors::Get_Honors() const
+{
+    Utf8String str = "Honors";
+    return Get_Int(str, 0);
+}
+
+void Reset_Battle_Honor_Insertion()
+{
+    g_rowsToSkip = 0;
+}
+
+void Insert_Battle_Honor(GameWindow* listbox, Image const* image, char earned, int tooltip, int* row, int* column, Utf16String str, int score)
+{
+    int height = (int)(((float)g_theDisplay->Get_Height() / 600.0) * 41.0);
+    int width = (int)(((float)g_theDisplay->Get_Width() / 800.0) * 40.0);
```

Comment:
Are these meant to be double?

Reply:
You're right, `FILD` instruction is used in ZH